lib/Cake/View/Helper/HtmlHelper.php

```diff
@@ -829,6 +829,15 @@ public function image($path, $options = array()) {
 			unset($options['url']);
 		}
 
+		if (isset($options['base64']) && !$url) {
+			$fullPath = WWW_ROOT . $path;
+			$base64 = base64_encode(file_get_contents($fullPath));
+			$ext = pathinfo($fullPath, PATHINFO_EXTENSION);
+			$type = $this->response->getMimeType($ext);
```

Comment:
If you used the `File` object here, you could avoid touching the response, and have better error handling options when reading the file.